src/plugins/embeddable/public/lib/panel/embeddable_panel.tsx

```diff
@@ -83,6 +83,8 @@ interface Props {
   SavedObjectFinder: React.ComponentType<any>;
   stateTransfer?: EmbeddableStateTransfer;
   hideHeader?: boolean;
+  isRetained?: boolean;
+  isBorderless?: boolean;
```

Comment:
Did we agree on this change? The source issue only talks about context menus. This changes how the embeddable can be used and the API of embeddables. I would not club this change with this PR. Can we open a separate issue to decide on it first and then make the change?

FYI, i'm not against the feature, but rather, I want to document the new API better so that others are aware of it. e.g. I have no idea what `isRetained` signifies  unless i see this PR.

Reply:
I have not spoken with anyone about these changes. The `isBorderless` prop is helpful in order to display the embeddable panel without any extra styling, which is not desired in our upcoming feature. The `isRetained` prop prevents the embeddable from being destroyed—I will rename it to `isDestroyPrevented`, which might be a better name—my goal with the naming is to allow the default to be `false`. I will also add a comment above the prop for explanation.

As for grouping these changes together—for our feature, all of these changes are needed together. I generally find it _ok_ to group changes, but please let me know if it is required to split them into multiple PRs.